src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -63,17 +72,19 @@ export async function downloadBuildDependencies(buildDependencies: BuildDependen
       }
       commands.push(`mkdir -p ${targetDir}`);
       await Promise.all(
-        dependencyInfo.dependencies.map((dep) => {
+        dependencyInfo.dependencies.map(async (dep) => {
           try {
             executeCliCommand({
               command: 'curl',
-              args: ['-SfL', dep.url, '-o', `${targetDir}/${dep.filename}`],
+              args: ['--max-time', '10', '-SfL', dep.url, '-o', `${targetDir}/${dep.filename}`],
+              options: options,
             });
           } catch (error) {
             console.error(
               `ERROR! Could not curl ${dep.url} into ${targetDir}/${dep.filename}. Dependency information: `,
               dependencyInfo
             );
+            return commands;
           }
           commands.push(`curl -SfL ${dep.url} -o ${targetDir}/${dep.filename}`);
         })
```

Comment:
Hmm, interesting thing here. I could be wrong, but I ***believe*** that because we're returning a static value (even though it's a Promise because of the async keyword) it can return before the `executeCliCommand` function finishes. 

I think this can be slightly changed to do what you're trying to do and cleaner too. If you take away the `async` keyword and add `return` keyword to the `executeCliCommand` invocation that will ensure that the Promise.all waits for each Promise to finish. Then move the `commands.push` line up into the try block before the return. Then you can completely delete the unused line `return commands`. Doing this will ensure that the Node thread will not continue on until all of these commands have either finished or errored.

I have an example below.

Like this: 
```
await Promise.all(
     dependencyInfo.dependencies.map((dep) => {
          try {
            commands.push(`curl -SfL ${dep.url} -o ${targetDir}/${dep.filename}`);
            return executeCliCommand({
              command: 'curl',
               args: ['--max-time', '10', '-SfL', dep.url, '-o', `${targetDir}/${dep.filename}`],
              options: options,
            });
          } catch (error) {
            console.error(
              `ERROR! Could not curl ${dep.url} into ${targetDir}/${dep.filename}. Dependency information: `,
              dependencyInfo
            );
          }
       })
)
```

Reply:
These are all great points, @mmeigs. The one small nit with the example though is that we won't hit this catch block, I think. This is because we return the promise, but we don't await it, not giving the promise the opportunity to settle before the catch block can be entered. 

I think we can add the `async` back and do `return await executeCliCommand...` so that it will wait for the promise to settle before returning.

This also has me thinking as well, do we want to do an `allSettled` here maybe as well? That way, we can know all of the dependencies that are failing for whatever reason.

Reply:
Testing this out with some incorrect curl links but this approach still doesn't seem to catch the error 😞

Reply:
Hmm, I wonder if doing `return await executeCliCommand(...)` we try:

```ts
 const cliResult = await executeCliCommand(...);

return cliResult;
```

in the try/catch? I wonder if return is sort of overwriting the `await` in a sense